h/services/annotation_stats.py

```diff
@@ -38,13 +41,18 @@ def group_annotation_count(self, pubid):
         """
         Return the count of shared annotations for this group.
         """
+        search = Search(self.request, stats=self.request.stats)
+        search.clear()
+        search.append_modifier(query.Limiter())
+        search.append_modifier(query.AuthFilter(self.request))
+        search.append_modifier(query.DeletedFilter())
```

Comment:
When I view http://localhost:5000/groups/__world__/public on my system, I see a much higher number for the annotation count than I do on master. Dumping the executed search queries using `search.to_dict` in `h.search.Search._search`, I see the following query executed:

```json
{
  "query": {
    "bool": {
      "filter": [
        {
          "bool": {
            "must_not": [
              {
                "exists": {
                  "field": "deleted"
                }
              }
            ]
          }
        },
        {
          "bool": {
            "should": [
              {
                "term": {
                  "shared": true
                }
              },
              {
                "term": {
                  "user_raw": "acct:robert@localhost"
                }
              }
            ]
          }
        }
      ]
    }
  },
  "sort": [
    {
      "updated": {
        "order": "desc",
        "unmapped_type": "boolean"
      }
    }
  ],
  "from": 0,
  "size": 0,
  "_source": false
}
```

I can see a few problems with this:

1. Annotations are not actually filtered by group.
2. Replies are not filtered out
3. Moderated and nipsa'd annotations are not filtered out

If I delete the `search.clear()` and `append_modifier` lines then the returned count is consistent with master.

Reply:
The way this query was written before on the db, none of those things were true before. Note that more annotations are shown in the group count from the db on the right than there are in the annotation count from search. 
![image](https://user-images.githubusercontent.com/30059933/46360360-121ddc00-c620-11e8-9131-806e22e09094.png)

That's not to say we can't change it. Personally I think it's a bit confusing as I would expect those numbers to match but I was trying to keep as close as possible to the existing behavior. The only intended difference is the logged in user's private annotations are included in the count.

I think the one issue here as Sean pointed out-I durpely overlooked attaching the GroupFilter and the tests didn't catch it cause they don't create another group.